File: Assignment/Abstraction/DeclarationNode.cs
namespace Assignment.Abstraction
{
    internal struct DeclarationNode: INode
    {
        public DeclarationNode(string name)
        {
            this.Name = name;
        }

        public string Name { get; }

        public string DisplayName()
        {
            return $"{nameof(DeclarationNode)}(name: {this.Name})";
        }
    }
}

File: src/main/java/ce305/abstraction/expressions/DeclarationNode.java
package ce305.abstraction.expressions;

import ce305.abstraction.DataType;
import ce305.abstraction.INode;

public class DeclarationNode implements INode {
    public final String name;
    public final DataType dataType;

    public DeclarationNode(String name, DataType dataType) {
        super();
        this.name = name;
        this.dataType = dataType;
    }

    @Override
    public String getDisplayName() {
        return String.format("%s(name: %s, type: %s)", DeclarationNode.class.getSimpleName(), this.name, this.dataType);
    }
}